pypesto/collections/collections.py

```diff
@@ -0,0 +1,497 @@
+        User needs to hand over a predictor function and settings, then all
+        results are grouped as CollectionPrediction for the whole ensemble
+
+        Parameters
+        ----------
+        predictor:
+            Prediction function, e.g., an AmiciPredictor
+        prediction_id:
+            Identifier for the predictions
+        sensi_orders:
+            Specifies which sensitivities to compute, e.g. (0,1) -> fval, grad.
+        mode:
+            Whether to compute function values or residuals.
+
+        Returns
+        -------
+        result:
+            CollectionPrediction of the collection object for the predictor
+            function
+        """
+        # preallocate
+        prediction_results = []
+
+        for ix in range(self.n_vectors):
+            x = self.x_vectors[:, ix]
+            prediction_results.append(predictor(x, sensi_orders, mode))
+
+        return CollectionPrediction(predictor=predictor,
+                                    prediction_id=prediction_id,
+                                    prediction_results=prediction_results)
+
+    def compute_summary(self,
+                        percentiles_list: Sequence[int] = (5, 20, 80, 95)):
+        """
+        This function computes the mean, the median, the standard deviation
+        and possibly percentiles for the parameters of the collection.
+        Those summary results are added as a data member to the
+        CollectionPrediction object.
+
+        Parameters
+        ----------
+        percentiles_list:
+            List or tuple of percent numbers for the percentiles
+
+        Returns
+        -------
+        summary:
+            Dict with mean, std, median, and percentiles of parameter vectors
+        """
+        # compute summaries based on parameters
+        summary = {MEAN: np.mean(self.x_vectors, axis=1),
+                   STANDARD_DEVIATION: np.std(self.x_vectors, axis=1),
+                   MEDIAN: np.median(self.x_vectors, axis=1)}
+        for perc in percentiles_list:
+            summary[f'{PERCENTILE} {perc}'] = np.percentile(self.x_vectors,
+                                                            perc, axis=1)
+        # store and return results
+        self.summary = summary
+        return summary
+
+    def check_identifiability(self) -> pd.DataFrame:
+        """
+        This functions uses ensemble mean and standard deviation to assess
+        (in a rudimentary way) whether or not parameters are identifiable.
+        It returns a dataframe with tuples, which specify whether or not the
+        lower and the upper bound are violated
+
+        Returns
+        -------
+        parameter_identifiability:
+            DataFrame indicating parameter identifiability based on mean
+            plus/minus standard deviations and parameter bounds
+        """
+        # first check if summary was computed. If not, do so
+        if self.summary is None:
```

Comment:
this won't update if the user changes the underlying collection after running check_identifiability the first time.

Reply:
Right... Question is whether one wants to leave this to the user... Probably not. The computational effort to compute this is minor, so we can really do this every time...